apps/backend/src/app/modules/maps/leaderboard-handler.util.spec.ts

```diff
@@ -57,27 +57,38 @@ describe('LeaderboardHandler', () => {
     });
   });
 
-  describe('getStageLeaderboardSuggestions', () => {
-    it('should return stage leaderboard suggestions', () => {
+  describe('getStageLeaderboard', () => {
+    it('should return stage leaderboards', () => {
       expect(
         LeaderboardHandler['getStageLeaderboards'](
           [
-            { gamemode: Gamemode.AHOP, trackType: TrackType.MAIN, trackNum: 0 },
+            { gamemode: Gamemode.AHOP, trackType: TrackType.MAIN, trackNum: 1 },
             {
               gamemode: Gamemode.BHOP,
               trackType: TrackType.BONUS,
-              trackNum: 0
+              trackNum: 1
             },
-            { gamemode: Gamemode.BHOP, trackType: TrackType.MAIN, trackNum: 0 }
+            { gamemode: Gamemode.BHOP, trackType: TrackType.MAIN, trackNum: 1 }
           ],
           ZonesStub
         )
       ).toMatchObject([
-        { gamemode: Gamemode.AHOP, trackType: TrackType.STAGE, trackNum: 0 },
         { gamemode: Gamemode.AHOP, trackType: TrackType.STAGE, trackNum: 1 },
-        { gamemode: Gamemode.BHOP, trackType: TrackType.STAGE, trackNum: 0 },
-        { gamemode: Gamemode.BHOP, trackType: TrackType.STAGE, trackNum: 1 }
+        { gamemode: Gamemode.AHOP, trackType: TrackType.STAGE, trackNum: 2 },
```

Comment:
Maybe I'm not following this code correctly, but it shouldn't be possible to have a stage with trackNum = 2 (any track numbers above 1 are for bonuses).

Reply:
I don't follow. These zones have a main track with two segments so it generates two stages, w/ s1 having tracknum = 1, s2 tracknum = 2. Why would trackNum > 1 be bonuses?

Reply:
Is there a concept of "stage number" here? If not then maybe trackNum is ok to use for stages. When I rewrote zone maker, I only used trackNum = 0 for main and trackNum >= 1 for bonuses.

Reply:
`trackNum` isn't a part of zones, it's for tracks, timers and leaderboards. A map with a two segment main track and two bonuses would have the following tracks:
```
trackType: TrackType.MAIN,  trackNum: 1
trackType: TrackType.STAGE, trackNum: 1
trackType: TrackType.STAGE, trackNum: 2
trackType: TrackType.BONUS, trackNum: 1
trackType: TrackType.BONUS, trackNum: 2
````
I'm not sure what you're talking about when you talk about a "trackNum" in zone maker, isn't zone maker just generating zone json files? I'm so confused by this. How would we distinguish stage tracks and leaderboards if not for a variable like `trackNum`?

Reply:
Nick - I think it's okay for you to resolve this when you look at this PR. This usage of `trackNum`, as a variable used alongside `trackType` to define tracks (with the usage given above) is completely ubiquitous across both this codebase and the game timer code (see usage of `MomTimer::TrackID`). Happy to talk through this with Panzer at some point but don't think this should hold up this PR.